firedrake/formmanipulation.py

```diff
@@ -127,6 +128,56 @@ def argument(self, o):
                          for j in numpy.ndindex(V_is[i].value_shape)]
         return self._arg_cache.setdefault(o, as_vector(args))
 
+    def cofunction(self, o):
+        from firedrake import Cofunction, FunctionSpace, MixedFunctionSpace
+        from pyop2 import MixedDat
+        from ufl.classes import ZeroBaseForm
```

Comment:
These should be top-level imports. The `firedrake` ones may need to be here to avoid circular imports but ideally should be moved too.